screens/Menu/ShuttleBus.js

```diff
@@ -1,32 +1,245 @@
+        let currentMinutes = new Date().getMinutes();
+        //set current time 
+        setCurrentTime({ Hour: currentHour, Minutes: currentMinutes });
+
+        let CurrentDayIndex = new Date().getDay();
+        let days = [];
+        days = weekDays;
+        days.forEach((item, index) => {
+            if (index == CurrentDayIndex) {
+                //set current day
+                setCurrentDay(item);
+            }
+        });
+    };
+
+    /**
+     * Function that calculates the time difference between the current time and the times of the next
+     * shuttle bus stops
+     * @param {*} nextStop 
+     */
+    const calculateTimeDifference = (nextStop) => {
+        var diffHours = nextStop.hour - currentTime.Hour;
+        var diffMinutes = nextStop.minutes - currentTime.Minutes;
+        if (diffHours >= 1 && Math.abs(diffMinutes) < 10) {
+            return ({
+                Hours: diffHours + "h",
+                Minutes: "0" + Math.abs(diffMinutes) + "min"
+            });
+        }
+        else if (diffHours < 1) {
+            return ({
+                Hours: "",
+                Minutes: Math.abs(diffMinutes) + " min"
+            });
+        }
+        else if (Math.abs(diffMinutes) < 1) {
+            return ({
+                Hours: diffHours + " h",
+                Minutes: ""
+            });
+        }
+        else {
+            return ({
+                Hours: diffHours + "h",
+                Minutes: Math.abs(diffMinutes) + "min"
+            });
+        }
+
+    };
+
+    /**
+     * Function that returns an object containing the id, timeDifference as well as the time of the next buses on schedule
+     * @param {*} list 
+     */
+    const fetchNextStops = (list) => {
+        var nextStops = [];
+        var results = [];
+        let index = 0;
+        for (var key in list) {
+            if (list[key].hour >= currentTime.Hour && list[key].minutes >= currentTime.Minutes) {
+                nextStops = list.slice(index, list.length - 1);
+                break;
+            }
+            index++;
+        }
+        if (nextStops.length > 0) {
+            results = nextStops.map((element, index) => {
+                return ({
+                    id: index.toString(),
+                    timeDifference: calculateTimeDifference(element),
+                    hour: element.hour,
+                    minutes: element.minutes
+                });
+            });
+        }
+        return results;
+    };
+
+    /**
+     * Function that sorts through the shuttle bus schedule and compares the time/day to the current time/day
+     */
+    const getNextStops = () => {
+        var scheduleTimesSGW = [];
+        var scheduleTimesLoyola = [];
+        const sgwCampus = "SGW";
+        const loyolaCampus = "Loyola";
+        if (currentDay && weekDays.includes(currentDay.toString()) && !friday.includes(currentDay.toString()) && !weekends.includes(currentDay.toString())) {
```

Comment:
I can see bugs hiding in that if statement, so I would suggest to not check on arrays for fridays or weekends to validate that is a weekday... 
**My Suggestion:** 
Instead of that if condition, I would check **if** `CurrentDayIndex` is between 1 and 4, if yes, then it is mondayToThursday. **Else if** `CurrentDayIndex` is 5, then it's Friday. **Else** it's the weekend.

Reply:
Very good suggestion thank you!! I've refactored the code and code rid of all the hardcoded arrays :)